examples/thread-object-sharing/worker.js

```diff
@@ -0,0 +1,50 @@
+/** @type {import('./pkg/thread_object_sharing')} */
+import init, { OpaqueObject } from "./pkg/thread_object_sharing.js";
+
+/**
+ * @typedef {{type: 'load', memory: WebAssembly.Memory}} WorkerRequestLoad
+ * @typedef {{type: 'start', opaqueObjectPtr: number}} WorkerRequestStart
+ * @typedef {{type: 'stop'}} WorkerRequestStop
+ * @typedef {{type: 'increment', 'incrementVal': number}} WorkerRequestIncrement
+ * @typedef {WorkerRequestLoad | WorkerRequestStart | WorkerRequestStop | WorkerRequestIncrement} WorkerRequest
+ */
+
+/** @type {number | undefined} */
+let intervalTimer = undefined;
+let incrementVal = 10;
+
+/** @type {OpaqueObject} */
+const opaqueObject = Object.create(OpaqueObject.prototype);
+
+function tick() {
+    opaqueObject.add(incrementVal);
+}
+
+self.onmessage = async (event) => {
+    /**
+     * @type {WorkerRequest}
+     */
+    const request = event.data;
+
+    switch (request.type) {
+        case 'load':
+            await init(undefined, request.memory);
+            break;
+        case 'start':
+            opaqueObject.__wbg_ptr = request.opaqueObjectPtr;
```

Comment:
Is this really a supported use-case?
This is going to become the big question in this PR to begin with: can this be triggered with supported code in the first place?